server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -833,13 +834,15 @@ void buildSubAggs(SignificantStringTerms.Bucket[][] topBucketsPreOrd) throws IOE
 
         @Override
         SignificantStringTerms buildResult(long owningBucketOrd, long otherDocCount, SignificantStringTerms.Bucket[] topBuckets) {
+            // the subset size is missing for empty buckets (doc count of 0)
+            long subsetSize = owningBucketOrd < subsetSizes.size() ? subsetSizes.get(owningBucketOrd) : 0;
```

Comment:
Maybe `if the owningBucketOrd is not in the array that means the bucket is empty so the size has to be 0`?

Reply:
https://github.com/elastic/elasticsearch/pull/60823/commits/3ddfda379e81e9fb655cd872c0cf577fe87f89f8, thanks

Reply:
Thank you!